observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetryConfigurationConstants.java

```
@@ -46,16 +60,41 @@ private OpenTelemetryConfigurationConstants() {}
     public static final String EXPORTER_JAEGER = "jaeger";
     public static final String EXPORTER_OTLP = "otlp";
     public static final String DEFAULT_OTLP_ENDPOINT = "http://localhost:4317";
-    public static final int DEFAULT_BATCH_DELAY = 5000;
     public static final int DEFAULT_EXPORT_TIMEOUT = 30000;
+
+    public static final int DEFAULT_LOGS_EXPORT_INTERVAL = 1000;
+    public static final int DEFAULT_METRICS_EXPORT_INTERVAL = 60000;
+    public static final int DEFAULT_TRACE_EXPORT_INTERVAL = 5000;
+
     public static final int DEFAULT_MAX_QUEUE_SIZE = 2048;
     public static final int DEFAULT_MAX_EXPORT_BATCH_SIZE = 512;
 
+    public static final String EXEMPLAR_FILTER_ALWAYS_ON = "always_on";
+    public static final String EXEMPLAR_FILTER_ALWAYS_OFF = "always_off";
+    public static final String EXEMPLAR_FILTER_TRACE_BASED = "trace_based";
+
+    public static final String TEMPORALITY = "temporality";
+    public static final String TEMPORALITY_DELTA = "delta";
+    public static final String TEMPORALITY_LOWMEMORY = "lowmemory";
+    public static final String TEMPORALITY_CUMULATIVE = "cumulative";
+
+    public static final String HISTOGRAM_AGGREGATION = "histogram-aggregation";
+    public static final String AGGREGATION_EXPONENTIAL = "base2_exponential_bucket_histogram";
+    public static final String AGGREGATION_EXPLICIT = "explicit_bucket_histogram";
+
+    public static final String[] ALLOWED_HISTOGRAM_AGGREGATION = {AGGREGATION_EXPONENTIAL, AGGREGATION_EXPLICIT};
+    public static final String[] ALLOWED_EXEMPLAR_FILTERS = {EXEMPLAR_FILTER_ALWAYS_ON, EXEMPLAR_FILTER_ALWAYS_OFF, EXEMPLAR_FILTER_TRACE_BASED};
     public static final String[] ALLOWED_EXPORTERS = {EXPORTER_JAEGER, EXPORTER_OTLP};
```

This is somewhat unrelated to this PR, but maybe an opportunity to look at it.

In the OpenTelemetrySubsystemRegistrar#validateExporter() method it says that "The 'jaeger' exporter is no longer supported. Please update the system to use otlp".

Is it OK to still list "jaeger" as a valid option? It is offered as option in the Web Console, and it is even possible to set it via Web Console and CLI. But then Wildfly fails to reload / reboot. Perhaps it should throw an error during the write-attribute operation already, to not allow the subsystem to enter into invalid state..?

I can open a Jira for this (and work on it too), if it feels like something that should be fixed, but not in this PR.

---

I believe it's there for backwards compatibility, allowing older server configs to start and get a more meaningful failure, rather than just fall over due to, say, a parsing error. Does that sound accurate, @bstansberry ?

---

This is an interesting discussion but definitely should be handled elsewhere.

It's quite complicated and the answer is not to have the validator reject 'jaeger'.

Use cases:

1) Mixed domain. The 'jaeger' value may be valid for a host controller running previous version servers, so it cannot be rejected by the domain controller. Note this scenario also means the management api description (read-resource-description) means the 'jaeger' value is listed as valid. Because it is valid. (The r-r-d output doesn't try and discriminate between use cases.)

2) Migration. Failing boot of an admin-only server that is using a previously valid config leaves the user no option to correct their config except XML editing. (I mention admin-only here because running in admin-only is a kind of hint that config manipulation is what the user is doing. If the server isn't admin-only we have to assume the purpose of running it is to due normal request handling etc.)

Failing in Stage.MODEL when these conditions are not relevant is valid. The OperationContext provides any OSH with the data it needs to detect these cases. The #1 case being any "executing on a Host Controller" case; i.e. don't be trying to figure out that it's mixed domain. It's not possible for a subsystem to do that.

---

Looking more it sounds like OpenTelemetrySubsystemRegistrar#validateExporter() is doing the right thing.

---

If memory serves, you wrote it. :P